[Feat/Recuritment_serializer] recruitment_attachment_serializer 작성

moxopills opened the pull request:
## ✅ PR 요약
- 작업 요약: recruitement_attachment_serializer 작성

## 📄 상세 내용
- [x] 4개 시리얼라이저 구현 - 조회/생성/수정/리스트필드로 용도별 분리, ModelSerializer 기반으로 CRUD 최적화 및 DRF 표준 패턴 준수
- [x] 타입 안전성 및 검증 강화 - mypy 타입 힌트 완벽 적용, to_internal_value 오버라이딩으로 리스트 포맷/필드 존재/중복 3단계 검증
- [x] file_name/file_url 필수 검증, 최대 5개 제한(ListField max_length), 파일명 중복 방지 로직 구현

## 🧪 PR Checklist
- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

Commit:
✨ feat : recruitment_attachment_serializer 작성

2. 변경 또는 추가사항을 아래에 간략하게 작성하세요 ( 필수 )
recruitment_attachmnet 시리얼라이저 작성

wjdtjddns98 commented on apps/recruitment/serializers/recruitment_attachment_serializer.py:
class RecruitmentAttachmentSerializer(serializers.ModelSerializer[Any]):


    class Meta:
        model = RecruitmentAttachment
        fields = [
            "id",
            "file_name",
            "file_url",
            "created_at",
            "updated_at",
        ]
        read_only_fields = ["id"]
        
 


이부분에 read_only_fields 에 updated_at, created_at도 추가 해주시는게 DRF 관례에 적합해용

wjdtjddns98 commented on apps/recruitment/serializers/recruitment_attachment_serializer.py:
```diff
@@ -0,0 +1,98 @@
+            "file_url",
+        ]
+        extra_kwargs = {
+            "file_name": {"required": False},
```
여기부분 어차피 PATCH 요청시에 required=False가 기본이라 extra_kwargs 블록 제거로 코드 간결화 시킬수 있을 것 같아용

wjdtjddns98 commented on apps/recruitment/serializers/recruitment_attachment_serializer.py:
```diff
@@ -0,0 +1,98 @@
+    첨부파일 리스트 필드
+    """
+
+    child = serializers.DictField(child=serializers.CharField())
```
현재 DictField를 사용하고, to_internal_value 에서 수동으로 검증 하고 있는데 
RecruitmentAttachmentCreateSerializer와 필드가 같은 임시 시리얼라이저를 정의하여 child=TempAttachmentSerializer()로 사용하면, file_name의 max_length 및 required 검증 등을 DRF가 자동으로 처리 해줄 수 있습니다

개선방안이지 필수는 아닙니당

wjdtjddns98 reviewed (commented):
처음 두개 코멘트는 필수로 교체해주시고, 마지막 코멘트는 권장드립니다

moxopills commented:
넵! 알겠습니다!

Commit:
2. 변경 또는 추가사항을 아래에 간략하게 작성하세요 ( 필수 )
read_only_fields = updated,created 추가 extra_kwargs 제거 임시 시리얼라이저 생성후 정의

Commit:
💡 chore : isort,black 포멧팅

Meoyoug commented on apps/recruitment/serializers/recruitment_attachment_serializer.py:
```diff
@@ -0,0 +1,65 @@
+
+        processed = super().to_internal_value(data)
+
+        file_names = {item["file_name"] for item in processed}
```
KeyError의 위험이 있는데 get을 활용하는게 안전해보입니다.

Meoyoug commented on apps/recruitment/serializers/recruitment_attachment_serializer.py:
```diff
@@ -0,0 +1,65 @@
+    file_url = serializers.URLField(required=True)
+
+
+class RecruitmentAttachmentListField(serializers.ListField):
```
필드를 오버라이딩하려는 시도는 굉장히 좋습니다!
다만 입력데이터가 각 딕셔너리 객체가 나열된 list형태라면 굳이 이렇게 별도로 만들지않아도 `many=True` 라는 옵션만 추가했어도 괜찮을 것 같아요

moxopills replied:
넵 알겠습니다! 수정하겠습니다!

Commit:
💡 chore: get으로 KeyError 방지 수정했습니다.

2. 변경 또는 추가사항을 아래에 간략하게 작성하세요
리뷰 1: get으로 KeyError 방지 수정했습니다!

Commit:
💡 chore    : ListField 제거

 2. 변경 또는 추가사항을 아래에 간략하게 작성하세요 ( 필수 )
ListField 제거 하고 recruitment_serializer에 many=True 사용 예정

SWWS97 reviewed (changes requested):
대면으로 내용 전달했습니다

Commit:
💡 chore: 모델에 맞게 length 수정 update 로직 추가

Commit:
💡 chore    : isort, black 포멧팅

Commit:
💡 chore: Any제거
